tests/neg/LiquidClass1.hs

```diff
@@ -0,0 +1,39 @@
+module LiquidClass where
+
+
+-- | Typing classes
+-- | Step 1: Refine type dictionaries:
+
+class Compare a where
+	cmax :: a -> a -> a
+	cmin :: a -> a -> a
+
+instance Compare Int where	
+{-@ instance Compare Int where 
+	cmax :: Odd -> Odd -> Odd ;
```

Comment:
Where is this checked?

On Mon, Jan 12, 2015 at 11:05 AM, Eric Seidel <notifications@github.com>
wrote:

> In tests/neg/LiquidClass1.hs
> <https://github.com/ucsd-progsys/liquidhaskell/pull/304#discussion-diff-22812841>
> :
>
> > @@ -0,0 +1,39 @@
> > +module LiquidClass where
> > +
> > +
> > +-- | Typing classes
> > +-- | Step 1: Refine type dictionaries:
> > +
> > +class Compare a where
> > +	cmax :: a -> a -> a
> > +	cmin :: a -> a -> a
> > +
> > +instance Compare Int where	
> > +{-@ instance Compare Int where
> > +	cmax :: Odd -> Odd -> Odd ;
>
> Yep!
>
>
> https://github.com/ucsd-progsys/liquidhaskell/blob/master/include/GHC/Real.spec
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/pull/304/files#r22812841>.
>



-- 
Niki Vazou